### ISSUE TITLE
getReceipt occasionally returns `Invalid node AccountID was set for transaction: 0.0.0`

### ISSUE THREAD

**Issue by gregscullard** [hashgraph/hedera-sdk-go#292]
### Description

Submitting a TransferTransaction (on mainnet) and getting a receipt for the transaction occasionally throws a `Invalid node AccountID was set for transaction: 0.0.0` error.

### Steps to reproduce

The error is intermittent, however this is the code used when it occurs

```
	client := hedera.ClientForMainnet()
	client.SetOperator(fromAccountId, privateKey)

	transaction := hedera.NewTransferTransaction().
		AddHbarTransfer(fromAccountId, hedera.HbarFrom((-1 * amountFloat), hedera.HbarUnits.Tinybar)).
		AddHbarTransfer(toAccountId, hedera.HbarFrom(amountFloat, hedera.HbarUnits.Tinybar))

	transaction.SetTransactionMemo(memo)
	transaction.SetMaxTransactionFee(hedera.HbarFrom(maxFeeAmountFloat, hedera.HbarUnits.Tinybar))

	txResponse, err := transaction.Execute(client)
	if err != nil {
		logger.ErrorLog("SendToAddress transaction.Execute(client) err: "+err.Error())
		reply.Content.Error = err.Error()
		return err
	}

	transferReceipt, err := txResponse.GetReceipt(client)
	if err != nil {
		logger.ErrorLog("SendToAddress txResponse.GetReceipt(client) err: "+err.Error())
		reply.Content.Error = err.Error()
		return err
	}
```

### Hedera network

mainnet

### Version

v2.1.16

**Comment by danielakhterov** [comment]
We are in the middle of a `_Network` refactor which I think might solve this issue. So we'll put this to the side until #290 gets merged.

**Comment by failfmi** [comment]
Hello, any progress here?

**Comment by danielakhterov** [comment]
@failfmi v2.4.0 was released with our network refactor that was mentioned above. Have you encountered the same issue since then?

**Comment by failfmi** [comment]
We've encountered it for a couple of hours, only for the first 4 testnet nodes.

**Comment by danielakhterov** [comment]
Thanks for letting us know, we'll make this a top priority for next sprint.

**Comment by failfmi** [comment]
Currently, the same thing occurs on testnet.

**Comment by danielakhterov** [comment]
@gregscullard @failfmi I've taken a deep look at the SDK, and I just cannot seem to find where this error can pop up. I've been looking for where we somehow return an empty response **and** an empty error which I believe would be the cause of this issue. Since I couldn't find anything myself I decided to at least add some panic statements which should hopefully help determine from where a node account ID of `0.0.0` is being set. I released it as https://github.com/hashgraph/hedera-sdk-go/releases/tag/v2.5.1-beta.1 If you can please update to this release and let me see the relevant logs it'd be much appreciated.

**Comment by gregscullard** [comment]
@danielakhterov some feedback from another project (they have not yet tried 2.5.1beta.1)

```
I am using github.com/hashgraph/hedera-sdk-go/v2 v2.1.16.
I got 2 errors:
[{TransactionID:0.0.2548686@1637029009.205320487 ScheduledTransactionId: NodeID:0.0.0 Hash:[]}]
[exceptional precheck status INVALID_NODE_ACCOUNT received for transaction 0.0.2548686@1637029009.205320487]
```

**Comment by sam-at-luther** [comment]
@danielakhterov I have done some testing on 2.5.1-beta.1 but I believe I'm encountering a new error introduced in https://github.com/hashgraph/hedera-sdk-go/commit/b45b0306216c8d8585f75bb0ae426da09f355db5  which is blocking me from doing a comprehensive test. 

Specifically I cannot transfer tokens due to `TRANSFERS_NOT_ZERO_SUM_FOR_TOKEN `. It seems `GetTokenTransfers()` is incorrectly only looking for NFT transfers, even in the non-NFT token transfer case.

**Comment by danielakhterov** [comment]
@sam-at-luther #320 Getting @andrix10 on it asap

**Comment by danielakhterov** [comment]
@gregscullard Issue with that log is that v2.1.16 is pre network refactor

**Comment by sam-at-luther** [comment]
I am still seeing this on `github.com/hashgraph/hedera-sdk-go/v2 v2.5.1-beta.1.0.20211116221542-f924b6714d5e`

```
info, err := hedera.NewTokenInfoQuery().
		SetNodeAccountIDs([]hedera.AccountID{nodeAccountID}).
		SetTokenID(hederaTokenID).
		Execute(s.backend)
	if err != nil {
		return nil, fmt.Errorf("info query execute, node: [%s]: %w", nodeID, err)
	}
```

```
failed to get token details: info query execute, node: [0.0.5]: exceptional precheck status INVALID_NODE_ACCOUNT
```

**Comment by danielakhterov** [comment]
Oh, that is very weird. I'm guessing this is the result of the payment transaction's node account ID being different than the node the request was submitted two. That's a huge issue. @andrix10

**Comment by sam-at-luther** [comment]
We're still seeing this on `github.com/hashgraph/hedera-sdk-go/v2 v2.5.1-beta.1.0.20211118093217-b38341cdc85f`


```	newBalance := hedera.NewHbar(2)
	accountResp, err := hedera.NewAccountCreateTransaction().
		SetNodeAccountIDs([]hedera.AccountID{nodeAccountID}).
		SetKey(newKey.PublicKey()).
		SetInitialBalance(newBalance).
		Execute(s.backend)
	if err != nil {
		return nil, err
	}
	receipt, err := accountResp.GetReceipt(s.backend)
	if err != nil {
		return nil, err
	}
```

**Comment by danielakhterov** [comment]
Is there a stack trace for the panic? Also, I do believe we have a potential fix in progress #325

**Comment by sam-at-luther** [comment]
This was on `b38341`

```
goroutine 6 [running]:
testing.tRunner.func1.2({0x158b520, 0x177eff0})
        /usr/local/Cellar/go/1.17.1/libexec/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
        /usr/local/Cellar/go/1.17.1/libexec/src/testing/testing.go:1212 +0x218
panic({0x158b520, 0x177eff0})
        /usr/local/Cellar/go/1.17.1/libexec/src/runtime/panic.go:1038 +0x215
github.com/hashgraph/hedera-sdk-go/v2.(*Query).SetNodeAccountIDs(0xc000299040, {0xc00008ab20, 0x0, 0x0})
        /Users/swood/work/pkg/mod/github.com/hashgraph/hedera-sdk-go/v2@v2.5.1-beta.1.0.20211118093217-b38341cdc85f/query.go:48 +0x92
github.com/hashgraph/hedera-sdk-go/v2.(*TransactionReceiptQuery).SetNodeAccountIDs(...)
        /Users/swood/work/pkg/mod/github.com/hashgraph/hedera-sdk-go/v2@v2.5.1-beta.1.0.20211118093217-b38341cdc85f/transaction_receipt_query.go:165
github.com/hashgraph/hedera-sdk-go/v2.TransactionResponse.GetReceipt({{0xc000618380, 0xc0006aaed0, 0x0}, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
        /Users/swood/work/pkg/mod/github.com/hashgraph/hedera-sdk-go/v2@v2.5.1-beta.1.0.20211118093217-b38341cdc85f/transaction_response.go:13 +0x1f3
```

**Comment by sam-at-luther** [comment]
I can reproduce that panic every time on `b3834` with the following test:
```
func TestBad(t *testing.T) {
	backend, err := hedera.ClientForName("testnet")
	require.NoError(t, err)

	operatorAccountID, err := AccountIDFromString(os.Getenv("HTS_OPERATOR_ID"))
	require.NoError(t, err)

	operatorKey, err := hedera.PrivateKeyFromString(os.Getenv("HTS_OPERATOR_KEY_SECRET"))
	require.NoError(t, err)

	backend.SetOperator(operatorAccountID, operatorKey)

	newKey, err := hedera.GeneratePrivateKey()
	require.NoError(t, err)

	nodeAccountID, err := AccountIDFromString("0.0.5")
	require.NoError(t, err)

	newBalance := hedera.NewHbar(2)
	accountResp, err := hedera.NewAccountCreateTransaction().
		SetNodeAccountIDs([]hedera.AccountID{nodeAccountID}).
		SetKey(newKey.PublicKey()).
		SetInitialBalance(newBalance).
		Execute(backend)
	require.NoError(t, err)

	_, err = accountResp.GetReceipt(backend)
	require.NoError(t, err)
}
```
Let me know if I'm doing something wrong.

**Comment by danielakhterov** [comment]
@sam-at-luther Thank you for the test and timing. Ran test and it consistently failed which allowed me to find out what was the issue. I'm pretty sure there were multiple issues that caused this, but the last one I found is simply that `errPersistent` was not being set accordingly which resulted in ```errors.Wrapf(errPersistent, "retry %d/%d", attempt, maxAttempts)``` returning `nil`. So that is how we somehow returned an empty `TransactionResponse{}` **and** `error`. @andrix10 please add this change to your PR and add the above test (after you rename it :stuck_out_tongue: )

**Comment by danielakhterov** [comment]
Update:
The change I made worked for me flawlessly, but for some reason it did not work 100% of the time for @andrix10 . I think we'll get what we have merged in as the fix _did_ work for me with your example, but we'll continue working on it as well to make sure it never happens again. One thing to note though, in Go `assert.NoError()` which is what we use (I assume `require` in your example is just a rename) does not stop the test progressing if there was in fact an error. This means that this error _can_ happen in a testing environment even when it wouldn't happen in a real environment. The only way to fix this is by returning or panicing on the error instead of asserting it.

**Comment by danielakhterov** [comment]
Were able to reproduce and fix this with some mocking: released in https://github.com/hashgraph/hedera-sdk-go/releases/tag/v2.5.1

**Comment by failfmi** [comment]
Currently, I am experiencing the same issue on testnet with version v2.5.1

**Comment by danielakhterov** [comment]
This is confirmed to be resolved in v2.6.0
